tests/rest/client/v1/test_login.py

```diff
@@ -273,9 +273,7 @@ def _delete_device(self, access_token, user_id, password, device_id):
 
         auth = {
             "type": "m.login.password",
-            # https://github.com/matrix-org/synapse/issues/5665
-            # "identifier": {"type": "m.id.user", "user": user_id},
-            "user": user_id,
+            "identifier": {"type": "m.id.user", "user": user_id},
```

Comment:
We likely still want a test for the legacy login type.

Reply:
Hm, so the `login` unittest function still uses the old method: https://github.com/matrix-org/synapse/blob/c2e1a2110fbe9ead26b4ecbb1afd504ed035a04d/tests/unittest.py#L512-L518 So it's definitely still tested.

But it may be a better idea to update this to the new format and just have a specific test for the old behaviour.